wordpress-mobile/WordPress-iOS
# Stats Revamp: StatsPeriodStore reduce superfluous onChange events

This PR adds a to StatsPeriodStore so that all the fetching states cause a single onChange trigger in a single transaction

The impact of this change is as follows:
 * currently when pressing a TimeInterval of Days/Weeks/Months etc on Stats screen, the onChange event is called 35 times
 * this transaction change reduces the onChange events to 15 times

Ref #17929

To test:
1. setup Xcode to log a message when breakpoint hit.  Example below.  I clicked Edit breakpoint + changed Action to Log Message.  Then I added `SiteStatsPeriodViewModel %B hit %H` and I checked the option at bottom to `Automatically continue after evaluating actions`

<img width="733" alt="image" src="https://user-images.githubusercontent.com/88816724/153569675-9b323f2d-9542-46f6-8340-841df43ca66b.png">


2. Go to Stats screen and then click then click the Days / Weeks / Months tab
App should behave normally

## Regression Notes
1. Potential unintended areas of impact
Stats Period screens

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Tested manually

3. What automated tests I added (or what prevented me from doing so)
None

PR submission checklist:
